File: src/main/java/jb/data/clfobj/ClfObjectDoubleSorted.java
package jb.data.clfobj;

public class ClfObjectDoubleSorted extends ClfObject {

    public ClfObjectDoubleSorted(double[] x, int y) {
        super(x, y);
    }

    @Override
    public int compareTo(ClfObject o) {
        return y != o.getY() ? Integer.compare(y, o.getY()) : Double.compare(x[0], o.getX()[0]);
    }

    public ClfObjectOnceSorted convertToOnceSorted() {
        return new ClfObjectOnceSorted(this.getX(), this.getY());
    }
}

File: src/main/java/jb/data/clfobj/ClfObjectDoubleSorted.kt
package jb.data.clfobj

open class ClfObjectDoubleSorted(x: DoubleArray, y: Int) : ClfObject(x, y) {

    override fun compareTo(o: ClfObject): Int {
        return if (y != o.y) Integer.compare(y, o.y) else java.lang.Double.compare(x[0], o.x[0])
    }

    fun convertToOnceSorted(): ClfObjectOnceSorted {
        return ClfObjectOnceSorted(this.x, this.y)
    }
}
